rbd/group_snap_info.go

```diff
@@ -0,0 +1,155 @@
+//go:build ceph_preview
+
+package rbd
+
+/*
+#cgo LDFLAGS: -lrbd
+#include <errno.h>
+#include <stdlib.h>
+#include <rbd/librbd.h>
+
+// exact copy of librbd.h:rbd_group_snap_namespace_type_t as it is not included in all versions
+#ifndef RBD_GROUP_SNAP_NAMESPACE_TYPE_USER
```

Comment:
Instead of defining the underscored versions below, could `ifndef` block just be extended?

Reply:
Oh, yes, that probably works too. Would make it a little cleaner. I'll try that once the Ceph PR is merged.

Reply:
Unfortunately it is not possible to use `#ifdef` on something that isn't a macro. All copied constants/types now have the `_` prefix.

Reply:
Having API users work around it like this is wrong.  I think we should add `LIBRBD_SUPPORTS_GROUP_SNAP_GET_INFO` define in `librbd.h`.

Reply:
https://github.com/ceph/ceph/pull/59959

Reply:
Note that it is still required to include the typedefs in case `LIBRBD_SUPPORTS_GROUP_SNAP_GET_INFO` isn't defined. Without them, it becomes more difficult to write wrappers for `dlsym()` and parse the returned/allocated structures.